crates/matrix-sdk-base/src/error.rs

```diff
@@ -56,4 +56,8 @@ pub enum Error {
     #[cfg(feature = "e2e-encryption")]
     #[error(transparent)]
     MegolmError(#[from] MegolmError),
+
+    /// An error caused by calling matrix-rust-sdk functions with invalid parameters
+    #[error("matrix-rust-sdk function was called with invalid parameters")]
+    ApiMisuse,
```

Comment:
We'd need to be much more precise than that, because there's absolutely no way to understand what goes wrong, if looking at a rageshake that contains this error. I'd be fine with renaming this error `InvalidReceiveMembersParameters`.